crates/sui-core/src/authority.rs

```diff
@@ -847,6 +850,23 @@ impl AuthorityState {
             Ok(res) => res,
         };
 
+        if let Some(expected_effects_digest) = expected_effects_digest {
+            if effects.digest() != expected_effects_digest {
+                error!(
```

Comment:
Why not have everything in the panic message?

Reply:
panic messages often do not get formatted very well, being smeared throughout other log output, so putting the full effects in the panic message may make quite a mess.

Reply:
Note that because logging pipeline is async we will likely not see this error log at all (as the node will panic before logging will have a chance to process the event emitted here)